jumpscale/core/base/fields.py

```diff
@@ -105,5 +106,163 @@ def validate(self, value):
             self.field.validate(item)
 
 
+class Email(Field):
+    def __init__(self, default="", **kwargs):
+        super().__init__(default, **kwargs)
+        self.regex = r"^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$"
+
+    def validate(self, value):
+        """Check whether provided value is a valid email representation
+        Args:
+            value (str)
+        Returns:
+            Boolean expresion"""
+        super().validate(value)
+        return re.search(self.regex, value) is not None
+
+
+class Path(Field):
+    def __init__(self, default="", **kwargs):
+        super().__init__(default, **kwargs)
+        self.regex = r"^(/[^/ ]*)+/?$"
```

Comment:
let's see if pathlib can provide more, because this only works for linux

Reply:
`pathlib` does not have function to check the path is valid or not